apps/shipments/views/shipment.py

```diff
@@ -0,0 +1,164 @@
+"""
+Copyright 2019 ShipChain, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import json
+import logging
+from string import Template
+
+from django.conf import settings
+from django.core.exceptions import ObjectDoesNotExist
+from django.db.models import Q
+from django.http import HttpResponse
+from django.utils.decorators import method_decorator
+from django_filters.rest_framework import DjangoFilterBackend
+from fancy_cache import cache_page
+from influxdb_metrics.loader import log_metric
+from rest_framework import viewsets, permissions, status, filters
+from rest_framework.decorators import action
+from rest_framework.exceptions import NotFound, PermissionDenied
+from rest_framework.response import Response
+from shipchain_common.authentication import get_jwt_from_request
+
+from apps.jobs.models import JobState
+from apps.permissions import owner_access_filter, get_owner_id
+from ..filters import ShipmentFilter, SHIPMENT_SEARCH_FIELDS
+from ..geojson import render_filtered_point_features
+from ..models import Shipment, TrackingData, PermissionLink
+from ..permissions import IsOwnerOrShared
+from ..serializers import ShipmentSerializer, ShipmentCreateSerializer, ShipmentUpdateSerializer, ShipmentTxSerializer
+
+LOG = logging.getLogger('transmission')
+
+
+class ShipmentViewSet(viewsets.ModelViewSet):
+    queryset = Shipment.objects.all()
+    serializer_class = ShipmentSerializer
+    permission_classes = ((IsOwnerOrShared,) if settings.PROFILES_ENABLED
+                          else (permissions.AllowAny,))
+    filter_backends = (filters.SearchFilter, filters.OrderingFilter, DjangoFilterBackend,)
+    filterset_class = ShipmentFilter
+    search_fields = SHIPMENT_SEARCH_FIELDS
+    ordering_fields = ('updated_at', 'created_at', 'pickup_est', 'delivery_est')
+    http_method_names = ['get', 'post', 'delete', 'patch']
+
+    def get_queryset(self):     # pylint: disable=too-many-branches
```

Comment:
I also would like to understand why this pylint error is triggered when it was not previously